src/value/owned.rs

```diff
@@ -68,6 +68,38 @@ impl OwnedSymbolToken {
             source,
         }
     }
+
+    /// Decorates the [`OwnedSymbolToken`] with text.
+    pub fn with_text<T: Into<Rc<str>>>(mut self, text: T) -> Self {
+        self.text = Some(text.into());
+        self
+    }
+
+    /// Decorates the [`OwnedSymbolToken`] with a local ID.
+    pub fn with_local_sid(mut self, local_sid: SymbolId) -> Self {
+        self.local_sid = Some(local_sid);
+        self
+    }
+
+    /// Decorates the [`OwnedSymbolToken`] with an [`OwnedImportSource`].
+    pub fn with_source<T: Into<Rc<str>>>(mut self, table: T, sid: SymbolId) -> Self {
+        self.source = Some(OwnedImportSource::new(table, sid));
+        self
+    }
+}
+
+/// Constructs an [`OwnedSymbolToken`] with unknown text and a local ID.
+/// A common case for binary parsing (though technically relevant in text).
+#[inline]
+pub fn local_sid_tok(local_sid: SymbolId) -> OwnedSymbolToken {
+    OwnedSymbolToken::new(None, Some(local_sid), None)
+}
+
+/// Constructs an [`OwnedSymbolToken`] with just text.
+/// A common case for text and synthesizing tokens.
+#[inline]
+pub fn text_tok<T: Into<Rc<str>>>(text: T) -> OwnedSymbolToken {
+    OwnedSymbolToken::new(Some(text.into()), None, None)
 }
```

Comment:
Notice that we cannot construct a token without a `local_sid` or `text`.  Also note that the mutators only add something, never remove.  This is intentional, though it makes `$0` the odd-case that cannot be constructed, but we can add a function to produce that case.

Reply:
See #159

Reply:
> `$0`... cannot be constructed ... but we can add a function to produce that case.

👍